leshoya.rb

```diff
@@ -0,0 +1,176 @@
+# PRODUCT HAS A NAME, DIAMETER, MASS, MOONS, GRAVITY, AND DISTANCE FROM THE SUN VALUES
+class Planet #always camel case
+
+  attr_reader :name, :diameter, :mass, :moons, :gravity, :distance_from_the_sun
+  attr_writer :name, :diameter, :mass, :moons, :gravity, :distance_from_the_sun
+
+
+  def initialize (name, diameter, mass, moons,gravity, distance_from_the_sun)
+
+    @name=name
+    @diameter=diameter
+    @mass=mass
+    @moons=moons
+    @gravity=gravity
+    @distance_from_the_sun=distance_from_the_sun
+
+  end
+
+end
+
+# ​
+#   def initialize(attrs) #2
+#     # attrs = {name: "Earth", mass: 9838}
+#     @name = attrs[:name] #3
+#     @diameter = attrs[:diameter]
+#   end
+# ​
+#   def name #5
+#     @name #6
+#   end
+# ​
+#   def diameter
+#     @diameter
+#   end
+# end
+
+# # **SETTER METHOD ABOVE WHEN YOU PUT IN =NEW_NAME ***DONT DELETE
+
+puts "Choose a Planet From the List:Mercury, Venus, or Earth"
+command=gets.chomp
+
+
+earth =   Planet.new("earth", "12", "18", "24", "40", "20 miles")
```

Comment:
It looks like down on line 136 you were using a `Hash` for the arguments, but here you are using a bunch of different arguments, was this to try different approaches? Do you have any questions about either approach?